loopy/symbolic.py

```diff
@@ -1974,6 +1976,10 @@ def simplify_using_aff(kernel, expr):
             .get_inames_domain(inames)
             .project_out_except(inames, [dim_type.set]))
 
+    if non_inames:
+        extra_domain = isl.BasicSet("{[" + ",".join(non_inames) + "]}")
```

Comment:
Is there a way to do this better?

Reply:
This looks like a valid approach IMO. Could be made slightly more sophisticated:
- If expression contains calls/subscripts then either handling those or leaving a TODO for later.
- `space = space.add_dims(isl.dim_type.param, len(non_inames))` and then setting the names of dims to those in `non_inames`.

Reply:
(Thanks! :slightly_smiling_face:)